### PR TITLE
add ontonotes_conll dataset

### PR DESCRIPTION
**Pull request by richarddwang**
# Introduction of the dataset
OntoNotes v5.0 is the final version of OntoNotes corpus, and is a large-scale, multi-genre,
multilingual corpus manually annotated with syntactic, semantic and discourse information.

This dataset is the version of OntoNotes v5.0 extended and used in the CoNLL-2012 shared task
, includes v4 train/dev and v9 test data for English/Chinese/Arabic and corrected version v12 train/dev/test data (English only).

This dataset is widely used in name entity recognition, coreference resolution, and semantic role labeling.

In dataset loading script, I modify and use the code of [AllenNLP/Ontonotes](https://docs.allennlp.org/models/main/models/common/ontonotes/#ontonotes) to read the special conll files while don't get extra package dependency. 

# Some workarounds I did

1. task ids
I add tasks that I can't find anywhere `semantic-role-labeling`, `lemmatization`, and `word-sense-disambiguation` to the task category `structure-prediction`, because they are related to "syntax". I feel there is another good name for the task category since some tasks mentioned aren't related to structure, but I have no good idea.

2. `dl_manage.extract`
Since we'll get another zip after unzip the downloaded zip data, I have to use `dl_manager.extract` directly inside `_generate_examples`. But when testing dummy data, `dl_manager.extract` do nothing. So I make a conditional such that it  manually  extract data when testing dummy data.

# Help
Don't know how to fix the building doc error.

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Awesome ! Thanks for adding the dataset :)

The dataset card and dataset script are already really good, thanks ! I left a few comments:

File: datasets/ontonotesv5_conll2012/ontonotesv5_conll2012.py [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "document_id": datasets.Value("string"),
+                    "sentences": [
+                        {
+                            "part_id": datasets.Value("int32"),
+                            "words": datasets.Sequence(datasets.Value("string")),
+                            "pos_tags": datasets.Sequence(datasets.Value("string")),
```
[the remark this answers is not in the document]
Comment (richarddwang) [reply datasets/ontonotesv5_conll2012/ontonotesv5_conll2012.py:160]: Sure! I will take a look at the tag set.

File: datasets/ontonotesv5_conll2012/ontonotesv5_conll2012.py [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+        lang = self.config.language
+        conll_version = self.config.conll_version
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_zip = os.path.join(dl_dir, "zmycy7t9h9-1/conll-2012.zip")
+        if isinstance(dl_manager, datasets.MockDownloadManager):
+            import shutil
+
+            ext_dir = os.path.join(dl_dir, "zmycy7t9h9-1/extraction_hash")
+            shutil.unpack_archive(data_zip, ext_dir)
+        else:
```
[the remark this answers is not in the document]
Comment (richarddwang) [reply datasets/ontonotesv5_conll2012/ontonotesv5_conll2012.py:192]: No worries, I can try a quick fix for this issue. Thanks for your kindness.

File: datasets/ontonotesv5_conll2012/README.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+datasets.ClassLabel(num_classes=37, names=[
+  "O",
+  "B-PERSON", "B-NORP", "B-FAC", "B-ORG", "B-GPE", "B-LOC", "B-PRODUCT", "B-EVENT", "B-WORK_OF_ART", "B-LAW", "B-LANGUAGE", "B-DATE", "B-TIME", "B-PERCENT", "B-MONEY", "B-QUANTITY", "B-ORDINAL", "B-CARDINAL",
+  "I-PERSON", "I-NORP", "I-FAC", "I-ORG", "I-GPE", "I-LOC", "I-PRODUCT", "I-EVENT", "I-WORK_OF_ART", "I-LAW", "I-LANGUAGE", "I-DATE", "I-TIME", "I-PERCENT", "I-MONEY", "I-QUANTITY", "I-ORDINAL", "I-CARDINAL",
+])
+```
+
+### Data Fields
+
+Every element (sentence) under the "sentences" data field of every entry, is stored in the format as same as [AllenNLP/OntonotesSentence](https://docs.allennlp.org/models/main/models/common/ontonotes/#ontonotessentence), except for the following
```
[the remark this answers is not in the document]
Comment (richarddwang) [reply datasets/ontonotesv5_conll2012/README.md:137]: You're right, that will be much more readable.

File: datasets/conll2012_ontonotesv5/conll2012_ontonotesv5.py [hunk 4]
```diff
@@ -0,0 +1,819 @@
...
+            ),
+            homepage="https://conll.cemantix.org/2012/introduction.html",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        lang = self.config.language
+        conll_version = self.config.conll_version
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_zip = glob(f"{dl_dir}/**/conll-2012*", recursive=True)[0]
```
Comment (lhoestq) [inline datasets/conll2012_ontonotesv5/conll2012_ontonotesv5.py:156]:
```suggestion
        data_zip = glob(os.path.join(dl_dir, "**/conll-2012*"), recursive=True)[0]
```

File: datasets/conll2012_ontonotesv5/conll2012_ontonotesv5.py [hunk 5]
```diff
@@ -0,0 +1,819 @@
...
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"conll_files_directory": os.path.join(data_dir, f"test/data/{lang}")},
+            ),
+        ]
+
+    def _generate_examples(self, conll_files_directory):
+        """Yields examples."""
+        conll_files = sorted(glob(f"{conll_files_directory}/**/*gold_conll", recursive=True))
```
Comment (lhoestq) [inline datasets/conll2012_ontonotesv5/conll2012_ontonotesv5.py:177]:
```suggestion
        conll_files = sorted(glob(os.path.join(conll_files_directory), "**/*gold_conll"), recursive=True))
```

**Review by lhoestq — APPROVED** [review]
Thank you ! Just did some minor changes

**Comment by lhoestq** [comment]
The CI fail is unrelated to this dataset, merging :)

**Pull request merged**